test/libyul/yulSyntaxTests/loadimmutable_shadowing.yul

```diff
@@ -0,0 +1,7 @@
+{
+    function loadimmutable(a) {}
+}
+// ====
+// dialect: evm
+// ----
+// ParserError 5568: (15-28): Cannot use builtin function name "loadimmutable" as identifier name.
```

Comment:
As @ekpyron perhaps this should not happen in the EVM dialect, only in the EVM object dialect.

However as it turned out, the `evm` dialect in yul syntax tests refers to evm with objects: https://github.com/ethereum/solidity/blob/develop/test/libyul/Common.cpp#L115-L117

This was changed in a5ae51fa6e (#8442) where during refactoring it was to enable objects. I'm not sure if it was planned or not, @a3d4 @chriseth ?

Reply:
We need to find a way to have both: decent coverage for the inline assembly dialect (we only have very few actual solidity tests using inline assembly) and the object dialect - at best without duplicating or triplicating all of them... we probably need to have most tests run against multiple dialects and then a few ones running against only a subset or something like that...

Reply:
Arguably, the builtin errors shouldn't be caught by the parser anyways, but rather as an analysis step... then most parser tests could be shared (that is except for dot-identifier-tests :-D)

Reply:
> This was changed in a5ae51f (#8442) where during refactoring it was to enable objects. I'm not sure if it was planned or not, @a3d4 @chriseth ?

Not sure anymore, but I think it is just a side effect of refactoring. Should I revert?

Reply:
No, we have a discussion issue under #9620 where we can sort this out later.